crates/ibc/src/core/ics04_channel/handler/chan_open_init.rs

```diff
@@ -10,11 +10,12 @@ use crate::core::ics04_channel::events::OpenInit;
 use crate::core::ics04_channel::msgs::chan_open_init::MsgChannelOpenInit;
 use crate::core::ics24_host::identifier::ChannelId;
 use crate::core::ics24_host::path::{ChannelEndPath, SeqAckPath, SeqRecvPath, SeqSendPath};
-use crate::core::router::ModuleId;
+use crate::core::router::{ModuleId, Router};
 use crate::core::{ContextError, ExecutionContext, ValidationContext};
 
 pub(crate) fn chan_open_init_validate<ValCtx>(
     ctx_a: &ValCtx,
+    router_a: &impl Router,
     module_id: ModuleId,
```

Comment:
> Though, I see the latter way might not be as straightforward to implement because of the dynamic dispatch for modules.

Opened this: https://github.com/cosmos/ibc-rs/issues/804